src/Common/src/Interop/OSX/Interop.libproc.cs

```diff
@@ -0,0 +1,523 @@
+
+internal static partial class Interop
+{
+    internal static partial class libproc
+    {
+        // Constants from sys\param.h
+        private const int MAXCOMLEN = 16;
+        private const int MAXPATHLEN = 1024;
+        
+        // Constants from proc_info.h
+        private const int MAXTHREADNAMESIZE = 64;
+        private const int PROC_PIDLISTFDS = 1;
+        private const int PROC_PIDTASKALLINFO = 2;
+        private const int PROC_PIDTHREADINFO = 5;
+        private const int PROC_PIDLISTTHREADS = 6;
+        private const int PROC_PIDPATHINFO_MAXSIZE = 4 * MAXPATHLEN;
+        private static int PROC_PIDLISTFD_SIZE = Marshal.SizeOf<proc_fdinfo>();
+        private static int PROC_PIDLISTTHREADS_SIZE = (Marshal.SizeOf<uint>() * 2);
+        
+        // Constants from sys\resource.h
+        private const int RUSAGE_SELF = 0;
+
+        // Defines from proc_info.h
+        internal enum ThreadRunState
+        {
+            TH_STATE_RUNNING            = 1,
+            TH_STATE_STOPPED            = 2,
+            TH_STATE_WAITING            = 3,
+            TH_STATE_UNINTERRUPTIBLE    = 4,
+            TH_STATE_HALTED             = 5
+        }
+
+        // Defines in proc_info.h
+        [Flags]
+        internal enum ThreadFlags
+        {
+            TH_FLAGS_SWAPPED    = 0x1,
+            TH_FLAGS_IDLE       = 0x2
+        }
+
+        /// <summary>
+        /// Converts an OS-level thread state to a .NET thread state.
+        /// </summary>
+        /// <param name="state">The OS state to convert</param>
+        /// <returns>The .NET mapping of the OS thread state</returns>
+        internal static System.Diagnostics.ThreadState ConvertOsxThreadRunStateToThreadState(ThreadRunState state)
+        {
+            switch (state)
+            {
+                case ThreadRunState.TH_STATE_RUNNING:
+                    return System.Diagnostics.ThreadState.Running;
+                case ThreadRunState.TH_STATE_STOPPED:
+                    return System.Diagnostics.ThreadState.Terminated;
+                case ThreadRunState.TH_STATE_HALTED:
+                    return System.Diagnostics.ThreadState.Wait;
+                case ThreadRunState.TH_STATE_UNINTERRUPTIBLE:
+                    return System.Diagnostics.ThreadState.Running;
+                case ThreadRunState.TH_STATE_WAITING:
+                    return System.Diagnostics.ThreadState.Standby;
+                default:
+                    throw new ArgumentOutOfRangeException("state");
+            }
+        }
+
+        /// <summary>
+        /// Convert an OS-level thread flag to a .NET ThreadWaitReason
+        /// </summary>
+        /// <param name="flags">The OS flags to convert</param>
+        /// <returns>
+        /// Returns a single thread wait reasons based on flag priority.
+        /// </returns>
+        internal static System.Diagnostics.ThreadWaitReason ConvertOsxThreadFlagsToWaitReason(ThreadFlags flags)
```

Comment:
Ditto.  These are less about interop signatures / imports / etc. and more about how we surface the information through the APIs we expose.